README.md

```diff
@@ -1,3 +1,215 @@
+# RTC instructions
+
+The runtimecore branch of this project will include any custom checks and tools that are not part of the upstream
+llvm-project. These changes cannot easily be fast-forwarded as they are generally based on a release of llvm which
+diverges from the main trunk every release. Once RTC upgrades, a branch called runtimecore_X.X.X will be created
+symbolicating the llvm-project release that was being used at the time. Sometimes an upgrade will only comprise the
+linux compiler used. Other branches may be for the the llvm tools which are needed on all platforms. It is best to check
+all the different runtimecore branches.
+
+The linux clang compiler is also a custom build. This is due to the need of using libc++ and libc++abi instead of
+libstdc++. Since some of our products link to libstdc++, it is crucial to statically link in these runtimes and to hide
+all symbols in order to allow using both runtimes in one application.
+
+To get setup for the build, clone the right sources:
+
+```sh
+mkdir -p ${HOME}/llvm && cd ${HOME}/llvm
```

Comment:
Perhaps this should be `mkdir -p /llvm && cd /llvm`?

Reply:
Not quite here. This is supposed to be done outside of the container and also applies to macOS. Windows would just be `/c/llvm` so I can add a comment there. The idea is that you'd use your ssh keys to clone the source outside of the container but then mount you local workspace into the container's `/llvm` directory. This is done on the docker run command below with `--volume ${HOME}/llvm:/llvm` which does the mounting in the container that is run without a user and doesn't have a ${HOME}.

I'm thinking about macOS and the idea of removing the user path from the build sounds like a good one. When things segfault, I notice they show up with my name on macOS since that is the path to the original source. I just don't know of a good place to put the source outside of the user's folder aside from something like `/tmp`, which would be dangerous to use as a reboot of the machine might wipe out everything.

Reply:
Got it. Thanks for explaining. For macOS, you could consider just creating a new user with a generic name for building the official compiler.

Reply:
I think I'll swap to the shared user at `/Users/Shared`. That seems to be built into the OS already. Linux doesn't have that but since we're mounting anyway, it won't matter there.